pkg/secrets/flags.go

```diff
@@ -38,11 +42,16 @@ func (o *CLIOptions) Validate() []error {
 	} else if o.BwUser != "" {
 		credentialsProviderConfigured = append(credentialsProviderConfigured, "bitwarden")
 	}
-
-	if vals := sets.NewString(o.VaultAddr, o.VaultTokenFile, o.VaultPrefix); len(vals) != 1 && ((len(vals) != 3) || vals.Has("")) {
-		errs = append(errs, fmt.Errorf("--vault-addr, --vault-token and --vault-prefix must be specified together"))
-	} else if len(vals) == 3 {
+	b2i := func(b bool) int {
```

Comment:
Why does anyone use this silly language?

Reply:
can we instead just if any of addr, token or prefix is set verify that all are set? This is not good code.

Reply:
(and I am aware that I started it)

Reply:
That is the verification it performs (see tests).

Other than the ridiculous step required in Go to have access to fundamental operations, I don't see what is not good about it.  It checks the required condition, and gets reasonably close to maximum efficiency:

<details>

``` go
	return b2i(o.s0 != "") + b2i(o.s1 != "" || o.s2 != "") + b2i(o.s3 != "")
```

```asm
        movq    "".o+8(SP), AX
        movq    8(AX), CX
        cmpq    24(AX), $0
        jeq     _O_f_pc61
        testq   CX, CX
        movl    $1, DX
_O_f_pc24:
        movq    56(AX), AX
        testq   CX, CX
        setne   CL
        testq   AX, AX
        setne   AL
        movblzx DL, DX
        movblzx CL, CX
        addq    DX, CX
        movblzx AL, AX
        addq    CX, AX
        movq    AX, "".~r0+16(SP)
        ret
_O_f_pc61:
        cmpq    40(AX), $0
        setne   DL
        testq   CX, CX
        jmp     _O_f_pc24
```

</details>

https://godbolt.org/z/PfnqK9vxn